app/database/models.py

```diff
@@ -172,26 +173,86 @@ class PSQLEnvironmentError(Exception):
     )
 
 
+class InvitationStatusEnum(enum.Enum):
+    unread = 0
+    accepted = 1
+    declined = 2
+
+
 class Invitation(Base):
     __tablename__ = "invitations"
 
     id = Column(Integer, primary_key=True, index=True)
-    status = Column(String, nullable=False, default="unread")
-    recipient_id = Column(Integer, ForeignKey("users.id"))
-    event_id = Column(Integer, ForeignKey("events.id"))
     creation = Column(DateTime, default=datetime.now)
+    status = Column(
+        Enum(InvitationStatusEnum),
+        default=InvitationStatusEnum.unread,
+        nullable=False,
+    )
 
+    recipient_id = Column(Integer, ForeignKey("users.id"))
+    event_id = Column(Integer, ForeignKey("events.id"))
     recipient = relationship("User")
     event = relationship("Event")
 
+    def decline(self, session: Session) -> None:
+        """declines the invitation."""
+        self.status = InvitationStatusEnum.declined
+        session.add(self)
+        session.commit()
+
+    def accept(self, session: Session) -> None:
+        """Accepts the invitation by creating an
+        UserEvent association that represents
+        participantship at the event."""
+
+        association = UserEvent(
+            user_id=self.recipient.id,
+            event_id=self.event.id
+        )
+        self.status = InvitationStatusEnum.accepted
+        session.add(self)
+        session.add(association)
+        session.commit()
```

Comment:
What about changing from decline to accept and vice versa?

Reply:
I agree that the user needs an option to change his mind, and I will create a `notification-history` page, where the user can change his choies.
but I still think that it will be best if the user could leve the event from the `view event` page.

Reply:
I'm not sure I understand your point

Reply:
I will try to explain my self beter.
I am now working on a page that shows all the old notifications.
In this page, the user can view old messages and accept declined invitations.
In my opinion, the user should not have an option to leave the event from this page, only from the event page.
what to you think?

Reply:
We're currently talking about the Backend implementation.
You add a new entry for the user accepting or declining event, but technically, the current architecture won't allow me to change the response of the user from "accepted" to "declined"